libnetdata/dyn_conf/dyn_conf.c

```diff
@@ -0,0 +1,906 @@
+            }
+            closedir(dir);
+        }
+        buffer_free(path);
+    }
+
+    dictionary_set(plugin->modules, module->name, module, sizeof(module));
+
+    return 0;
+}
+
+
+void handle_dyncfg_root(struct uni_http_response *resp, int method)
+{
+    if (method != HTTP_METHOD_GET) {
+        resp->content = "method not allowed";
+        resp->content_length = strlen(resp->content);
+        resp->status = HTTP_RESP_METHOD_NOT_ALLOWED;
+        return;
+    }
+    json_object *obj = get_list_of_plugins_json();
+    json_object *wrapper = json_object_new_object();
+    json_object_object_add(wrapper, "configurable_plugins", obj);
+    resp->content = strdupz(json_object_to_json_string_ext(wrapper, JSON_C_TO_STRING_PRETTY));
+    json_object_put(wrapper);
+    resp->status = HTTP_RESP_OK;
+    resp->content_type = CT_APPLICATION_JSON;
+    resp->content_free = freez;
+    resp->content_length = strlen(resp->content);
+}
+
+void handle_plugin_root(struct uni_http_response *resp, int method, struct configurable_plugin *plugin, void *post_payload, size_t post_payload_size)
+{
+    switch(method) {
+        case HTTP_METHOD_GET:
+        {
+            dyncfg_config_t cfg = plugin->get_config_cb(plugin->cb_usr_ctx);
+            resp->content = mallocz(cfg.data_size);
+            memcpy(resp->content, cfg.data, cfg.data_size);
+            resp->status = HTTP_RESP_OK;
+            resp->content_free = free;
+            resp->content_length = cfg.data_size;
+            return;
+        }
+        case HTTP_METHOD_PUT:
+        {
+            const char *response;
+            if (post_payload == NULL) {
+                resp->content = "no payload";
+                resp->content_length = strlen(resp->content);
+                resp->status = HTTP_RESP_BAD_REQUEST;
+                return;
+            }
+            dyncfg_config_t cont = {
+                .data = post_payload,
+                .data_size = post_payload_size
+            };
+            response = set_plugin_config(plugin, cont);
+            if (response == NULL) {
+                resp->status = HTTP_RESP_OK;
+                resp->content = "OK";
+                resp->content_length = strlen(resp->content);
+            } else {
+                resp->status = HTTP_RESP_BAD_REQUEST;
+                resp->content = response;
+                resp->content_length = strlen(resp->content);
+            }
+            return;
+        }
+        default:
+            resp->content = "method not allowed";
+            resp->content_length = strlen(resp->content);
+            resp->status = HTTP_RESP_METHOD_NOT_ALLOWED;
+            return;
+    }
+}
+
+void handle_module_root(struct uni_http_response *resp, int method, struct configurable_plugin *plugin, const char *module, void *post_payload, size_t post_payload_size)
+{
+    if (strncmp(module, DYN_CONF_SCHEMA, strlen(DYN_CONF_SCHEMA)) == 0) {
```

Comment:
Considering that `DYN_CONF_SCHEM` is a macro, I suggest to use `sizeof` instead `strlen` here, of course the same applies for the rest of the code.

Reply:
what is the benefit of that?

Reply:
Considering that we have a constant value, the compiler can determine its size, so we do not need to measure.

Reply:
You are right, it is replaced with a constant. Let us move ahead, because this is minor an your PR is working as expected. :)